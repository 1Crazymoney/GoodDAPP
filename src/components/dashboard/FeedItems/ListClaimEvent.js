// @flow
import React from 'react'
import { Icon } from 'react-native-elements'
import { Avatar } from 'react-native-paper'
import { Text, View } from 'react-native-web'
import BigGoodDollar from '../../common/BigGoodDollar'
import { listStyles } from './EventStyles'
import type { FeedEventProps } from './EventProps'

/**
 * Render list claim item for feed list
 * @param {FeedEventProps} feedEvent - feed event
 * @returns {HTMLElement}
 */
const ListClaimEvent = ({ item: feed }: FeedEventProps) => {
  return (
    <View style={listStyles.innerRow}>
      <View>
        <Avatar.Image size={48} style={{ backgroundColor: 'white' }} source={feed.data.endpoint.avatar} />
        <Text>{`\n`}</Text>
      </View>
      <View style={listStyles.rowData}>
        <Text style={listStyles.rowDataText}>Claimed</Text>
      </View>
      <View style={[listStyles.row, { borderBottomWidth: 0, marginBottom: 0, padding: 0 }]}>
        <BigGoodDollar number={feed.data.amount} elementStyles={listStyles.currency} />
        <View style={listStyles.rowData}>
          <Icon raised color="rgb(85, 85, 85)" size={24} name="call-received" />
          <Text style={{ fontSize: '8px', color: '#4b4b4b', opacity: 0.8 }}>
            {new Date(feed.date).toLocaleString()}
          </Text>
        </View>
      </View>
    </View>
  )
}

export default ListClaimEvent
