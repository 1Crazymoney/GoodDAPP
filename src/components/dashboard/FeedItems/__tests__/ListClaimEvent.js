import React from 'react'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
import ListClaimEvent from '../ListClaimEvent'
import { generateFeedItemProps } from '../../__tests__/__util__'

describe('ListClaimEvent', () => {
  const props = generateFeedItemProps('withdraw')
  it('renders without errors', () => {
    const tree = renderer.create(<ListClaimEvent {...props} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<ListClaimEvent {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
