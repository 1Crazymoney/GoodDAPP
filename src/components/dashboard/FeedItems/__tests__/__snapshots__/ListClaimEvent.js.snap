// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListClaimEvent matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "horizontal",
      "WebkitBoxPack": "center",
      "WebkitFlexDirection": "row",
      "WebkitJustifyContent": "center",
      "alignItems": "center",
      "backgroundColor": "rgba(238,238,239,1.00)",
      "flexDirection": "row",
      "justifyContent": "center",
      "msFlexAlign": "center",
      "msFlexDirection": "row",
      "msFlexPack": "center",
      "width": "100%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,1.00)",
          "borderBottomLeftRadius": "24px",
          "borderBottomRightRadius": "24px",
          "borderTopLeftRadius": "24px",
          "borderTopRightRadius": "24px",
          "height": "48px",
          "width": "48px",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitFlexBasis": "auto",
            "borderBottomLeftRadius": "24px",
            "borderBottomRightRadius": "24px",
            "borderTopLeftRadius": "24px",
            "borderTopRightRadius": "24px",
            "flexBasis": "auto",
            "height": "48px",
            "msFlexPreferredSize": "auto",
            "overflowX": "hidden",
            "overflowY": "hidden",
            "width": "48px",
            "zIndex": 0,
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.00)",
              "backgroundImage": "url(\\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAkCAIAAAB0Xu9BAAAABGdBTUEAALGPC/xhBQAAAuNJREFUWEetmD1WHDEQhDdxRMYlnBFyBIccgdQhKVcgJeQMpE5JSTd2uqnvIGpVUqmm9TPrffD0eLMzUn+qVnXPwiFd/PP6eLh47v7EaazbmxsOxjhTT88z9hV7GoNF1cUCvN7TTPv/gf/+uQPm862MWTL6fff4HfDx4S79/oVAlAUwqOmYR0rnazuFnhfOy/ErMKkcBFOr1vOjUi2MFn4nuMil6OPh5eGANLhW3y6u3aH7ijEDCxgCvzFmimvc95TekZLyMSeJC68Bkw0kqUy1K87FlpGZqsGFCyqEtQNDdFUtFctTiuhnPKNysid/WFEFLE2O102XJdEE+8IgeuGsjeJyGHm/xHvQ3JtKVsGGp85g9rK6xMHtvHO9+WACYjk5vkVM6XQ6OZubCJvTfPicYPeHO2AKFl5NuF5UK1VDUbeLxh2BcRGKTQE3irHm3+vPj6cfCod50Eqv5QxtwBQUGhZhbrGVuRia1B4MNp6edwBxld2sl1splfHCwfsvCZfrCQyWmX10djjOlWJSSy3VQlS6LmfrgNvaieRWx1LZ6s9co+P0DLsy3OdLU3lWRclQsVcHJBcUQ0k9/WVVrmpRzYQzpgAdQcAXxZzUnFX3proannrYH+Vq6KkLi+UkarH09mC8YPr2RMWOlEqFkQClsykGEv7CqCUbXcG8+SaGvJ4a8d4y6epND+pEhxoN0vWUu5ntXlFb5/JT7JfJJqoTdy9u9qc7ax3xJRHqJLADWEl23cFWl4K9fvoaCJ2BHpmJ3s3z+O0U/DmzdMjB9alWZtg4e3yxzPa7lUR7nkvxLHO9+tvJX3mtSDpwX8GajB283I8R8a7D2MhUZr1iNWdny256yYLd52DwRYBtRMvE7rsmtxIUE+zLKQCDO4jlxB6CZ8M17GhuY+XTE8vNhQiIiSE82ZsGwk1pht4ZSpT0YVpon6EvevOXXH8JxVR78QzNuamupW/7UB7wO/+7sG5V4ekXb4cL5Lyv+4IAAAAASUVORK5CYII=\\")",
              "backgroundPosition": "center",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "cover",
              "bottom": "0px",
              "height": "100%",
              "left": "0px",
              "position": "absolute",
              "right": "0px",
              "top": "0px",
              "width": "100%",
              "zIndex": -1,
            }
          }
        />
        <img
          alt=""
          className="css-accessibilityImage-9pa8cd"
          draggable={false}
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAAAkCAIAAAB0Xu9BAAAABGdBTUEAALGPC/xhBQAAAuNJREFUWEetmD1WHDEQhDdxRMYlnBFyBIccgdQhKVcgJeQMpE5JSTd2uqnvIGpVUqmm9TPrffD0eLMzUn+qVnXPwiFd/PP6eLh47v7EaazbmxsOxjhTT88z9hV7GoNF1cUCvN7TTPv/gf/+uQPm862MWTL6fff4HfDx4S79/oVAlAUwqOmYR0rnazuFnhfOy/ErMKkcBFOr1vOjUi2MFn4nuMil6OPh5eGANLhW3y6u3aH7ijEDCxgCvzFmimvc95TekZLyMSeJC68Bkw0kqUy1K87FlpGZqsGFCyqEtQNDdFUtFctTiuhnPKNysid/WFEFLE2O102XJdEE+8IgeuGsjeJyGHm/xHvQ3JtKVsGGp85g9rK6xMHtvHO9+WACYjk5vkVM6XQ6OZubCJvTfPicYPeHO2AKFl5NuF5UK1VDUbeLxh2BcRGKTQE3irHm3+vPj6cfCod50Eqv5QxtwBQUGhZhbrGVuRia1B4MNp6edwBxld2sl1splfHCwfsvCZfrCQyWmX10djjOlWJSSy3VQlS6LmfrgNvaieRWx1LZ6s9co+P0DLsy3OdLU3lWRclQsVcHJBcUQ0k9/WVVrmpRzYQzpgAdQcAXxZzUnFX3proannrYH+Vq6KkLi+UkarH09mC8YPr2RMWOlEqFkQClsykGEv7CqCUbXcG8+SaGvJ4a8d4y6epND+pEhxoN0vWUu5ntXlFb5/JT7JfJJqoTdy9u9qc7ax3xJRHqJLADWEl23cFWl4K9fvoaCJ2BHpmJ3s3z+O0U/DmzdMjB9alWZtg4e3yxzPa7lUR7nkvxLHO9+tvJX3mtSDpwX8GajB283I8R8a7D2MhUZr1iNWdny256yYLd52DwRYBtRMvE7rsmtxIUE+zLKQCDO4jlxB6CZ8M17GhuY+XTE8vNhQiIiSE82ZsGwk1pht4ZSpT0YVpon6EvevOXXH8JxVR78QzNuamupW/7UB7wO/+7sG5V4ekXb4cL5Lyv+4IAAAAASUVORK5CYII="
        />
      </div>
    </div>
    <div
      className="css-text-76zvg2"
      dir="auto"
    >
      

    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitFlexBasis": "0%",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "msFlexNegative": 1,
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0%",
      }
    }
  >
    <div
      className="css-text-76zvg2"
      dir="auto"
      style={
        Object {
          "color": "rgba(75,75,75,1.00)",
          "fontSize": "15px",
          "textTransform": "capitalize",
        }
      }
    >
      Claimed
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "center",
        "WebkitFlexDirection": "row",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "backgroundColor": "rgba(238,238,239,1.00)",
        "borderBottomColor": "rgba(0,0,0,0.10)",
        "borderBottomWidth": "0px",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": "0px",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexPack": "center",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(85,85,85,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "12px",
            "textAlign": "right",
          }
        }
      >
        0.04
      </div>
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(85,85,85,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "12px",
            "textAlign": "right",
          }
        }
      >
        GD
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxPack": "center",
            "WebkitJustifyContent": "center",
            "alignItems": "center",
            "backgroundColor": "rgba(255,255,255,1.00)",
            "borderBottomLeftRadius": "28px",
            "borderBottomRightRadius": "28px",
            "borderTopLeftRadius": "28px",
            "borderTopRightRadius": "28px",
            "height": "52px",
            "justifyContent": "center",
            "marginBottom": "7px",
            "marginLeft": "7px",
            "marginRight": "7px",
            "marginTop": "7px",
            "msFlexAlign": "center",
            "msFlexPack": "center",
            "width": "52px",
          }
        }
      >
        <div
          className="css-text-76zvg2"
          dir="auto"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.00)",
              "color": "rgba(85,85,85,1.00)",
              "fontFamily": "MaterialIcons",
              "fontSize": "24px",
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          î‚µ
        </div>
      </div>
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(75,75,75,1.00)",
            "fontSize": "8px",
            "opacity": 0.8,
          }
        }
      >
        4/1/2019, 3:03:14 PM
      </div>
    </div>
  </div>
</div>
`;
